src/gtfs/write.rs

```diff
@@ -55,13 +55,26 @@ pub fn write_transfers(path: &path::Path, transfers: &Collection<NtfsTransfer>)
 pub fn write_agencies(
     path: &path::Path,
     networks: &CollectionWithId<objects::Network>,
+    ticketing_deep_links: &TicketingDeepLinks,
 ) -> Result<()> {
     info!("Writing agency.txt");
     let path = path.join("agency.txt");
     let mut wtr =
         csv::Writer::from_path(&path).with_context(|| format!("Error reading {:?}", path))?;
-    for n in networks.values() {
-        wtr.serialize(Agency::from(n))
+    for network in networks.values() {
+        let mut agency = Agency::from(network);
+        if !ticketing_deep_links.is_empty() {
+            agency.ticketing_deep_link_id = network
+                .fare_url
+                .as_ref()
+                .and_then(|fare_url| {
+                    ticketing_deep_links
+                        .get(fare_url)
+                        .map(|ticketing_deep_link| ticketing_deep_link.id.clone())
+                })
+                .or(Some("".to_string()));
```

Comment:
Might be more efficient... and more lazy.

```suggestion
                .or_else(|| Some(String::new()));
```

Reply:
Sorry this functionality is shameful. We are twisted by GMap .Sorry you read that 🫣